### ISSUE TITLE
False positive: unsupported-membership-test with Singleton

### ISSUE THREAD

**Issue by pehala** [PyCQA/pylint#4121]
### Steps to reproduce

```python
class _Singleton:
    instance: '_Singleton' = None

    def __contains__(self, item):
        return item in ["a"]


# pylint: disable=invalid-name
def Singleton() -> _Singleton:
    """Returns instance of Singleton"""
    if _Singleton.instance is None:
        _Singleton.instance = _Singleton()
    return _Singleton.instance

print("a" in Singleton())
print("b" in Singleton())
```
There are many different ways of creating Singleton in Python, and this one specifically confuses pylint.
### Current behavior
`E1135: Value 'Singleton()' doesn't support membership test (unsupported-membership-test)`

### Expected behavior
No error regarding `unsupported-membership-test`.

### pylint --version output
```
pylint 2.7.0
astroid 2.5
Python 3.7.9 (default, Aug 19 2020, 17:05:11) 
[GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]
```
